From 85edac88e565f5ea02ee7f2ec9f38da801068f52 Mon Sep 17 00:00:00 2001
From: Jae Hoon Kim <kimjae@chromium.org>
Date: Wed, 24 May 2023 07:37:42 +0000
Subject: [PATCH] Restore base::Value::SetTypeKey

This restores base::Value::Set[Bool|Int]Key for the packages which are
still using the deprecated functions.

---
 base/values.cc | 8 ++++++++
 base/values.h  | 4 ++++
 2 files changed, 12 insertions(+)

diff --git a/base/values.cc b/base/values.cc
index a776651656..b0c2e1d8d1 100644
--- a/base/values.cc
+++ b/base/values.cc
@@ -1196,6 +1196,14 @@ Value* Value::SetKey(StringPiece key, Value&& value) {
   return GetDict().Set(key, std::move(value));
 }
 
+Value* Value::SetBoolKey(StringPiece key, bool value) {
+  return GetDict().Set(key, value);
+}
+
+Value* Value::SetIntKey(StringPiece key, int value) {
+  return GetDict().Set(key, value);
+}
+
 Value* Value::SetStringKey(StringPiece key, StringPiece value) {
   return GetDict().Set(key, value);
 }
diff --git a/base/values.h b/base/values.h
index 37f8cd6bd0..25e611d839 100644
--- a/base/values.h
+++ b/base/values.h
@@ -786,6 +786,10 @@ class BASE_EXPORT GSL_OWNER Value {
   //
   // DEPRECATED: Prefer `Value::Dict::Set()`.
   Value* SetKey(StringPiece key, Value&& value);
+  // DEPRECATED: Prefer `Value::Dict::Set()`.
+  Value* SetBoolKey(StringPiece key, bool val);
+  // DEPRECATED: Prefer `Value::Dict::Set()`.
+  Value* SetIntKey(StringPiece key, int val);
 
   // `Set<Type>Key` looks up `key` in the underlying dictionary and associates a
   // corresponding Value() constructed from the second parameter. Compared to
-- 
2.40.1.698.g37aff9b760-goog

